src/ar/hentaislayer/src/eu/kanade/tachiyomi/extension/ar/hentaislayer/HentaiSlayer.kt

```diff
@@ -0,0 +1,245 @@
+                is StatusFilter -> url.addQueryParameter("status", filter.toUriPart())
+                is GenresFilter ->
+                    filter.state
+                        .filter { it.state }
+                        .forEach { url.addQueryParameter("genre[]", it.uriPart) }
+                else -> {}
+            }
+        }
+
+        url.addQueryParameter("page", page.toString())
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    // =========================== Manga Details ============================
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        document.selectFirst("main section")?.run {
+            selectFirst("img#manga-cover")?.run {
+                thumbnail_url = imgAttr()
+                title = attr("alt")
+            }
+            selectFirst("section > div:nth-child(1) > div:nth-child(1) > div:nth-child(2) > div:nth-child(2)")?.run {
+                status = parseStatus(select("a[href*='?status=']").text())
+                genre = select("a[href*='?type=']").text()
+                author = select("p:has(span:contains(المؤلف)) span:nth-child(2)").text()
+                artist = select("p:has(span:contains(الرسام)) span:nth-child(2)").text()
+            }
+            selectFirst("section > div:nth-child(1) > div:nth-child(2)")?.run {
+                select("h1").text().takeIf { it.isNotEmpty() }?.let {
+                    title = it
+                }
+                genre = select("a[href*='?genre=']")
+                    .map { it.text() }
+                    .let {
+                        listOf(genre) + it
+                    }
+                    .joinToString()
+                select("h2").text().takeIf { it.isNotEmpty() }?.let {
+                    description = "Alternative name: $it\n"
+                }
+            }
+            description += select("#description").text()
+        }
+    }
+
+    private fun parseStatus(status: String) = when {
+        status.contains("مستمر") -> SManga.ONGOING
+        status.contains("متوقف") -> SManga.CANCELLED
+        status.contains("مكتمل") -> SManga.COMPLETED
+        else -> SManga.UNKNOWN
+    }
+
+    // ============================== Chapters ==============================
+    override fun chapterListSelector() = "main section #chapters-list a#chapter-item"
+
+    override fun chapterFromElement(element: Element) = SChapter.create().apply {
+        setUrlWithoutDomain(element.attr("href"))
+        name = "\u061C" + element.select("#item-title").text() // Add unicode ARABIC LETTER MARK to ensure all titles are right to left
+
+        date_upload = parseRelativeDate(element.select("#item-title + span").text()) ?: Calendar.getInstance().timeInMillis
```

Comment:
```suggestion
        date_upload = parseRelativeDate(element.select("#item-title + span").text()) ?: 0L
```
app already handles 0L date as current